Xenobiology nerfs and fixes

SmArtKar opened the pull request:
<!-- Write **BELOW** The Headers and **ABOVE** The comments else it may not be viewable. -->
<!-- You can view Contributing.MD for a detailed description of the pull request process. -->

_**I am truely sorry, my fellow xenobiologist brothers and sisters, but sacrifices must be made**_

## About The Pull Request

Nobody can deny that xenobio is OP as fuck. Here's list of all changes:

1. Luminiscents now have to spend some of their slimey blood to activate their abilities. 100 for minor, 200 for major. This will not put you into dangerous blood volume zone, but you'll have to regenerate your blood after each cast so it's not completely free.
2. Adamantine armor now requires time to put on. Bug where red stabilized extract removed it's slowdown was also fixed(Because the armor is supposed to be heavy I made it so the slowdown applies while you carry it as well)
3. Industrial purples were really powerful because they generated autoinjectors that healed 75 damage per use for 5 plasma. Halved amount of regen jelly and doubled the plasma cost for them.
4. Bloodchiller now takes twice as much blood per shot.
5. Regenerative(purple) crossbreeds no longer fully adminheal. Instead, they only heal 25 damage, 50 organ damage, patch up bleeding wounds, remove stuns and other nasty effects while normalizing your blood amount and body temperature

## Why It's Good For The Game

Xenobio is too OP. Oh and yeah, some of it's shit was abusable

## Changelog
:cl:
balance: Luminiscents' abilities now cost blood. 100 for minor and 200 for major
balance: Industrial purple crossbreeds now produce weaker autoinjectors and require twice as much plasma
balance: Bloodchiller now requires twice as much blood
balance: Regenerative crossbreeds no longer fully adminheal
fix: Adamantine armor no longer can be sped up using stabilized red crossbreed
/:cl:

<!-- Both :cl:'s are required for the changelog to work! You can put your name to the right of the first :cl: if you want to overwrite your GitHub username as author ingame. -->
<!-- You can use multiple of the same prefix (they're only used for the icon ingame) and delete the unneeded ones. Despite some of the tags, changelogs should generally represent how a player might be affected by the changes rather than a summary of the PR's contents. -->

SmArtKar committed:
I am sorry, my fellow xenobiologist brothers and sisters

TemporalOroboros commented:
> 150 for minor, 250 for major.

Are you _entirely certain_ that these will not kill the user. Last I checked 250 is a little less than half of the blood a person has which should put the user solidly in "I am going to die from hypotension".

SmArtKar commented:
> > 150 for minor, 250 for major.
> 
> Are you _entirely certain_ that these will not kill the user. Last I checked 250 is a little less than half of the blood a person has which should put the user solidly in "I am going to die from hypotension".

Each player initially has 560u of blood(2000u maximum). If user has 560u of blood when he uses major ability(blood regen is 0.25 per tick) then he will be left with 310u, which is 1u less than what... aw fuck, I forgot that I wanted to make it 200u instead of 250

SmArtKar committed:
Cost change

SmArtKar commented:
With 200 it will leave them with low, but not critical blood volume which will get regenerated pretty quicky. User will only recieve light oxy damage right after the use, that's all

Fikou commented:
u should remove the cooldowns if you are adding a blood cost it will be funny to see people kill themselves by lack of blood

SmArtKar commented:
> u should remove the cooldowns if you are adding a blood cost it will be funny to see people kill themselves by lack of blood

that... that's totally not a good idea. Especially with golden cores

SmArtKar committed:
Makes minor purple actually regenerate your blood

TemporalOroboros commented:
You might want to take a look at the purple slime extract's minor ability. It's currently set to restore 50 blood and 50 nutrition which, given the new cost makes it convert 100 blood to 50 nutrition. This would appear to be more silver's wheelhouse. Given that you use nutrition to regenerate blood at a 5:1 RoI this could actually be useful to regenerate blood, but if you're low enough that that's a problem you don't want to sac the 100 in the first place and the time it will take will probably make it pointless anyhow.

TemporalOroboros commented:
> 
> 
> > u should remove the cooldowns if you are adding a blood cost it will be funny to see people kill themselves by lack of blood
> 
> that... that's totally not a good idea. Especially with golden cores

On one hand memes, on the other balance.

SmArtKar commented:
> You might want to take a look at the purple slime extract's minor ability. It's currently set to restore 50 blood and 50 nutrition which, given the new cost makes it convert 100 blood to 50 nutrition. This would appear to be more silver's wheelhouse. Given that you use nutrition to regenerate blood at a 5:1 RoI this could actually be useful to regenerate blood, but if you're low enough that that's a problem you don't want to sac the 100 in the first place and the time it will take will probably make it pointless anyhow.

I already worked around that, it keeps it's 50 blood regen(it regens 150 to compensate the cost). So yeah, the only thing left there is cooldown

TheBonded commented:
For the regen crossbreeds, will they still cure viruses and brain traumas? I liked that you could use them to cure permanent brain traumas, or use them to survive a bit longer from a virus that's killing everyone on station

SmArtKar commented:
> For the regen crossbreeds, will they still cure viruses and brain traumas? I liked that you could use them to cure permanent brain traumas, or use them to survive a bit longer from a virus that's killing everyone on station

nope

TheBonded commented:
> > For the regen crossbreeds, will they still cure viruses and brain traumas? I liked that you could use them to cure permanent brain traumas, or use them to survive a bit longer from a virus that's killing everyone on station
> 
> nope

in that case, would there be any chance to have it heal 50 of all damages? I think that for it costing 10 purple extracts, which could be used to make 1000u of regen jelly, healing only 25 of all types is a bit weak. I do agree nerfing it from an aheal is important, but I feel like taking it down to 25 is too much of a nerf and will make it never used

Krysonism commented:
blood chiller is almost never used.

CRITAWAKETS commented:
about adamantine, you can just use flight potion aswell to mitigate the slowdown

that said it requires rainbows and thus more effort so its kind-of a good thing

ATH1909 commented:
people use the megaman gun?

also, obligatory :walter:

moocowswag commented:
Last time I used adamantine armor with flight potion it still slowed me down and I had to use stabilized red.

Also the blood chiller is just a weaker more spammable cryokinesis spell that takes an armslot, it simply shouldnt be nerfed.

TheBonded commented:
@moocowswag the adamantine armour has no helmet specifically so you can't use it with a stable adamantine and fullaugs/nanites to become literally unkillable

SmArtKar commented:
> Last time I used adamantine armor with flight potion it still slowed me down and I had to use stabilized red so I think that "interaction" was fixed
> 
> Adamantine armor should also get a button that gives the user a helmet so it atleast fills its extreme niche of making you into an extremely slow and almost useless tank (Good for area denial I guess? But you wont be able to run from grenades / ranged weapons and you`ll still be 2 hit by batons)
> 
> Also the blood chiller is just a weaker more spammable cryokinesis spell that takes an armslot, it simply shouldnt be nerfed.
> 
> One last thing, why are only the benign/beneficial things that xenobio can do always under fire?
> 
> This change ignores the elephant in the room that I always see players use as an excuse to hate on xenobio in general and that is the gold / oil / extract and golem spam which never fails to take entire rounds hostage when they are present.
> 
> > > u should remove the cooldowns if you are adding a blood cost it will be funny to see people kill themselves by lack of blood
> > 
> > 
> > that... that's totally not a good idea. Especially with golden cores
> 
> Except golden major activation only spawns one entity at a time and at your position, its gimmicky compared to being able to gather dozens upon dozens of gold extracts and running around the station spamming them with plasma, each one spawning SEVERAL hostile mobs, they may not inherit your faction like gold major but there is an easy way around it with stabilized pink (which eventually bugs out and makes all normally hostile mobs neutral to you no matter what)
> 
> Also nerfing lumi abilities directly by making them cost blood should either be supplemented or replaced entirely by just giving slime people opportunity costs, let them live their hard to kill fantasy and take away their ability to use guns, they are slime people afterall, they are probably too squishy to be able to pull a trigger.

Okay, so: Adamantine armor gives you 80% protection from brute and completely reflects all laser and energy projectiles. Helmet is missing intentionaly, so your head becomes your weak point.

SmArtKar commented:
> > > > For the regen crossbreeds, will they still cure viruses and brain traumas? I liked that you could use them to cure permanent brain traumas, or use them to survive a bit longer from a virus that's killing everyone on station
> 
> > > nope
> 
> > in that case, would there be any chance to have it heal 50 of all damages? I think that for it costing 10 purple extracts, which could be used to make 1000u of regen jelly, healing only 25 of all types is a bit weak. I do agree nerfing it from an aheal is important, but I feel like taking it down to 25 is too much of a nerf and will make it never used
> 
> Don't they all still have their other effects? I mean its still a instant 25 HP in all types along with all the other stuff they do. I personally would rather see them heal more effectively one or two types of damage (get's people to make more types of slimes along with making them more interesting beyond their effects).
> 
> > Adamantine armor now requires time to put on. Bug where red stabilized extract removed it's slowdown was also fixed(Because the armor is supposed to be heavy I made it so the slowdown applies while you carry it as well)
> 
> Why the putting on time? #58432 got closed because that was generally seen as a bad idea and just added tedium.

If you remove putting on time you can just hop it between slots so you don't get the slowdown. Not cool.

SmArtKar committed:
Merge branch 'master' into i_am_sorry_brothers

ATH1909 commented:
> @moocowswag the adamantine armour has no helmet specifically so you can't use it with a stable adamantine and fullaugs/nanites to become literally unkillable

there's an effective armor cap of 90%, y'know.

ATH1909 commented:
and completely reflects all laser and energy projectiles.

it has a 40% chance to reflect anything the ablative trenchcoat can reflect. it cannot reflect bullets, for example.

moocowswag commented:
> @moocowswag the adamantine armour has no helmet specifically so you can't use it with a stable adamantine and fullaugs/nanites to become literally unkillable

Except the slowdown is horrid against stuncombat, and im almost sure the armor does nothing against the easy shove stun into cable cuff combo (which even nerve splicing/stun resist nanites struggles to help against if rng is not in your favor)

Yes hulk exists but that limits you to melee which (when combined with insanely low speed) makes you an incredibly immobile threat that is still vulnerable to explosives, armor piercing syringes, someone just building a wall around you, throwing items at you, walking away from you, gas flooding you with a flamethrower since you cant move out of the cloud quickly, etc.

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -572,16 +578,17 @@
+	if(species.current_extract && H.blood_volume >= blood_cost)
 		var/cooldown = species.current_extract.activate(H, species, activation_type)
 		species.extract_cooldown = world.time + cooldown
+		H.blood_volume -= blood_cost
```
does this mean that if you start one of the do_after() extract activations but that do_after() gets interrupted, you'll still have to pay its blood cost?

SmArtKar replied:
Yep

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -572,16 +578,17 @@
 	CHECK_DNA_AND_SPECIES(H)
 
-	if(species.current_extract)
-		species.extract_cooldown = world.time + 100
```
this line is actually important; it keeps you from spamming out do_after() extract activations

ATH1909 commented:
it should also be noted that, unlike slimepeople, luminescents and stargazers cannot use their nutrition level to regenerate their slime jelly unless they're low on slime jelly.

SmArtKar commented:
> it should also be noted that, unlike slimepeople, luminescents and stargazers cannot use their nutrition level to regenerate their slime jelly unless they're low on slime jelly.

Yeah, you need to use plasma for that

SmArtKar committed:
re-adds first cooldown check

ATH1909 commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -571,17 +577,21 @@
 	CHECK_DNA_AND_SPECIES(H)
-
+	
 	if(species.current_extract)
 		species.extract_cooldown = world.time + 100
+
+	if(species.current_extract && H.blood_volume >= blood_cost)
 		var/cooldown = species.current_extract.activate(H, species, activation_type)
```
```suggestion
	if(species.current_extract && H.blood_volume >= blood_cost)
		species.extract_cooldown = world.time + 100
		var/cooldown = species.current_extract.activate(H, species, activation_type)
```
otherwise, you'll get put on a 10 second cooldown if you try to use an extract while you don't have enough blood to pay its blood cost.

D4C-420 commented:
> If you remove putting on time you can just hop it between slots so you don't get the slowdown. Not cool.

Doesn't it only fit on the chest though and you added the fact it slows you down in hand? The thing is too bulky to fit in back packs.  Again adding a add self delay seems like it would still only be there just to make people suffer without any real reason.  I think the in hand slow down is enough because I can't think of a single suit with slowdown that needs a delay to be put on to prevent dodging the slow down.

CRITAWAKETS commented:
> there's an effective armor cap of 90%, y'know.

armor cap is a myth, there is no armor cap

SmArtKar committed:
Update code/modules/mob/living/carbon/human/species_types/jellypeople.dm

Co-authored-by: ATH1909 <42606352+ATH1909@users.noreply.github.com>

SmArtKar commented:
> > If you remove putting on time you can just hop it between slots so you don't get the slowdown. Not cool.
> 
> Doesn't it only fit on the chest though and you added the fact it slows you down in hand? The thing is too bulky to fit in back packs. Again adding a add self delay seems like it would still only be there just to make people suffer without any real reason. I think the in hand slow down is enough because I can't think of a single suit with slowdown that needs a delay to be put on to prevent dodging the slow down.

three words. Bag Of Holding.

SmArtKar commented:
also pulling the suit will prevent the slowdown

D4C-420 commented:
> three words. Bag Of Holding.

Fair enough. Still don't like the idea of having the precedent for adding time to put clothing on.  

> also pulling the suit will prevent the slowdown

Then maybe keep it consistent and add slowdown when pulling it?

IndieanaJones commented:
I never liked crossbreeds but you really nerfed the megaman gun of all things, which was already shit?  For what reason?

Watermelon914 commented:
Yeah, as some other people say in this PR, the big problem currently that I can agree with is with gold extracts producing sentient mobs. Magicarps, Chaos Magicarps, combat bears and more are extremely strong and being able to make them sentient makes them even stronger. If you are doing xenobiology nerfs, it'd be a good idea to also nerf these things.

The wumborian fugu gland being especially broken allowing simplemobs to smash reinforced walls and have ridiculous damage.

IndieanaJones commented:
> Yeah, as some other people say in this PR, the big problem currently that I can agree with is with gold extracts producing sentient mobs. Magicarps, Chaos Magicarps, combat bears and more are extremely strong and being able to make them sentient makes them even stronger. If you are doing xenobiology nerfs, it'd be a good idea to also nerf these things.
> 
> The wumborian fugu gland being especially broken allowing simplemobs to smash reinforced walls and have ridiculous damage.

I'd note that only one other person has complained about gold cores in this whole thread, and I'd vehemently fight against any crippling change to gold.  Gold/Admantine is one of the few not self-serving powers xenobio has, and only really becomes an issue if there's a lot of willing ghosts in the round, which could say something about the state of the round itself.  I've said this several times, but the proper change I'd like to see is for the monsters to actually have job applications for use around the station.

Watermelon914 commented:
>> snip
> 
> I'd note that only one other person has complained about gold cores in this whole thread, and I'd vehemently fight against any crippling change to gold. Gold/Admantine is one of the few not self-serving powers xenobio has, and only really becomes an issue if there's a lot of willing ghosts in the round, which could say something about the state of the round itself. I've said this several times, but the proper change I'd like to see is for the monsters to actually have job applications for use around the station so their creation isn't only purposed for smashing shit.

After looking further into the code, the problem only seems to stem from the wumborian fugu making gold slime simplemobs incredibly strong, allowing them to deal significantly more damage and smash through reinforced walls with no downsides.

ATH1909 commented:
> > there's an effective armor cap of 90%, y'know.
> 
> armor cap is a myth, there is no armor cap

it exists, though

ATH1909 commented:
maybe I should go digging to confirm that ranged attacks have an effective armor cap as well.

moocowswag commented:
I think fugu gland + sentient mobs are okay, I dont have anything against gold core + sentience, the problem is spammable non sentients.

Fugu glands arnt exactly mass produceable since youre reliant on rng and youre putting in effort on each individual mob that you put a fugu gland in.

Meanwhile you can just take a bio bag of 25 gold cores, run around injecting them with plasma, annnnd whala you just spawned over 100 hostile mobs throughout the station and slowed down the server.

Golems on the other hand are a whole other ball park and are a policy nightmare due to their nature of group cohesion and escalation which have caused alot of salty ooc after many rounds.

Watermelon914 commented:
> I think fugu gland + sentient mobs are okay, I dont have anything against gold core + sentience, the problem is spammable non sentients.
> 
> Fugu glands arnt exactly mass produceable since youre reliant on rng and youre putting in effort on each individual mob that you put a fugu gland in. (A single fugu'd mob is only worth like 3-4 mobs, maybe more if you gave them healing nanites)
> 
> Meanwhile you can just take a bio bag of 25 gold cores, run around injecting them with plasma, annnnd whala you just spawned over 100 hostile mobs throughout the station and slowed down the server.
> 
> Golems on the other hand are a whole other ball park and are a policy nightmare due to their nature of group cohesion and escalation which have caused alot of salty ooc after many rounds.
> 
> And Im not saying that they NEED to be changed im just saying they are WAY more impactful on a round than some guy walking around with adamantine armor so why hate on the armor.

This is like saying pre-nerf hulks aren't mass producible because getting it before was rare. That's the thing, it's not rare and there is no harm in just trying over and over again until you get it. Also 1 mob with fugu gland is already more powerful than most things, considering they can break through reinforced walls, have double damage, have boosted health.

I don't think it's fair to let xenobiology produce such creatures, especially considering there's no downside. Now look at genetics. Hulk pre-nerf used to be an absolute annoying piece of shit because they could break rwalls with no downsides and were generally just unbearable as fuck. It was nerfed because it was never supposed to be that easily strong. This is what fugu glands do, they create pre-nerf hulks that are actually stronger if you use them on the right mobs, because then you can deal 50 damage per hit, suffer no slowdowns or stuns and have way more health than a hulk would ever have.

Whenever you see xenobio mobs boosted by fugu glands, you can be sure it'll affect the round. Especially if they're made by a traitor.

moocowswag commented:
> I don't think it's fair to let xenobiology produce such creatures, especially considering there's no downside.

The time spent equipping one mob with a fugu gland instead of just pumping out more is the downside.

Nexofas commented:
Adamantine armor couldnt be speed potion'ed before though. Did I miss something?

SmArtKar commented:
> Adamantine armor couldnt be speed potion'ed before though. Did I miss something?

It could be sped up by using red stabilized crossbreed which removed ALL equipment slowdowns passively

OmegaDarkPotato commented:
> Each player initially has 560u of blood(2000u maximum). 

I... don't think this is the case. Each player has 560 cubic liters. 1 cl is not 1u, unless iv drips do wild and wacky shit. From my experience, it's more like "each player has 200 units of blood", since a single blood bag (100 units) brings someone from 50% to 100%

Watermelon914 commented:
> > I don't think it's fair to let xenobiology produce such creatures, especially considering there's no downside.
> 
> The time spent equipping one mob with a fugu gland instead of just pumping out more is the downside, I only ever bother with fugu glanding if ghosts are apparently in limited supply.

Not a lot of time. Comparable to getting hulk, it's all RNG.

SmArtKar commented:
> > Each player initially has 560u of blood(2000u maximum).
> 
> I... don't think this is the case. Each player has 560 cubic liters. 1 cl is not 1u, unless iv drips do wild and wacky shit. From my experience, it's more like "each player has 200 units of blood", since a single blood bag (100 units) brings someone from 50% to 100%

Maybe, that doesn't really matter in my case tho

SpaceVampire commented:
I know a couple of people said it already, but why would you even want to nerf the Bloodchiller?
If I had to name the top five most powerful xenobio things, it wouldn't be close.

SmArtKar committed:
Merge branch 'master' into i_am_sorry_brothers

SmArtKar committed:
oops

LemonInTheDark commented:
What are the fixes? you only really mention nerfs

SmArtKar commented:
> What are the fixes? you only really mention nerfs

adamantine armor one?

moocowswag commented:
Can we at least get more tweaks on the cost to extract activations?

Having to expend 600 blood per minute to use dark blue minor is kind of atrocious considering its an ability thats meant to be spammed to get any use out of.

AndroidMK2 commented:
The changes are rasional but please leave my luminescent boys the way they are ;-;

Kylerace commented on code/modules/mob/living/carbon/human/species_types/jellypeople.dm:
```diff
@@ -551,6 +551,12 @@
 	icon_icon = 'icons/mob/actions/actions_slime.dmi'
 	background_icon_state = "bg_alien"
 	var/activation_type = SLIME_ACTIVATE_MINOR
+	var/datum/species/jelly/luminescent/species
```
this var doesnt seem to be used and even if it was you dont clean it up if the object its referencing gets qdel'd. what are you trying to do here?

Kylerace commented on code/modules/research/xenobiology/crossbreeding/regenerative.dm:
```diff
@@ -23,17 +23,42 @@ Regenerative extracts:
+			humi.blood_volume = max(humi.blood_volume, BLOOD_VOLUME_NORMAL)
+
+		for(var/i in humi.internal_organs)
+			var/obj/item/organ/O = i
```
as anything here

Kylerace commented on code/modules/research/xenobiology/crossbreeding/regenerative.dm:
```diff
@@ -23,17 +23,42 @@ Regenerative extracts:
+
 	core_effect_before(H, user)
-	H.revive(full_heal = TRUE, admin_revive = FALSE)
+	activate_healing(H)
```
this revived in the past, if you dont want to allow that then put it in the changelog

Kylerace commented on code/modules/research/xenobiology/crossbreeding/regenerative.dm:
```diff
@@ -23,17 +23,42 @@ Regenerative extracts:
 	playsound(target, 'sound/effects/splat.ogg', 40, TRUE)
 	qdel(src)
 
+/obj/item/slimecross/regenerative/proc/activate_healing(mob/living/target) //No longer full adminheal
```
comments like this only make sense for people reading this pr, not people reading this code in the future, properly document this with that in mind https://github.com/tgstation/tgstation/blob/master/.github/AUTODOC_GUIDE.md

Learner10 commented:
Of all the Xenobio things I see, the blood chiller is not one of them. Basically, we're taking one of the least commonly seen tool a Xenobiologist has available for their use, balancing it by making it more likely to kill the user through the most likely cause of death for a longer-lived antagonist, aka blood loss, all because it was too good at doing what exactly?

Let's look it over rather quickly. 1. It takes up an entire arm slot permanently, which is a significant price of using it. 2. It drains the user of blood volume per use. 3. Each use of the blood chiller fires a projectile that... chills those it hits. I'm looking at the code for it, and I don't think the projectile even does direct damage! Plus, this also wouldn't effect two races, or simple mobs.

If anything... it _needs_ a buff to see it being used. If it were retractable, even if it applied hefty injury on the limb for each time it retracts (and only retracts), that would be a more reasonable balance change for one of the less useful things the department can produce.

CloverIsLucky commented:
Hello, I'm from a server which mirrors everythings from here, I saw this PR and I want to show my discomfort for these nerfs with a few suggestion and things:

1.- The purple slimes nerf seems stupidly nerfed, the rest of the things seems okay, but the only that has been annoying me is the xenobio regen cores wildly nerfed. You should make the 10 purple extracts worth and I have a few ideas for that thanks to some guys I've discussed my same discomforts about the nerf.

Reworking some of these purple crossbreeds: 
* Regenerative Grey Slime: No special effects, only heals 30-35 everything including organ damage, fully restores stamina and stabilizes bleeding wounds.
* Regenerative Orange Slime: Should be reworked from causing a ring of fire to only fully heal burn damage, unhusk bodies and fully removes burn wounds.
* Regenerative Purple: Fully heals brute damage and heals 30 damage of everything and the target is dosed with 8u of regenerative jelly.

The rest of the PR, seems a bit good, but in compensation of nerfing the stabilized red to not work with adamantine armor, you should add an adamatine helmet, so it could compensate the slowdown.

What do you think?
